java-checks/src/main/java/org/sonar/java/checks/unused/UnusedMethodParameterCheck.java

```diff
@@ -80,7 +89,34 @@ private static boolean isExcluded(MethodTree tree) {
   }
 
   private static boolean isAnnotated(MethodTree tree) {
-    return !tree.modifiers().annotations().isEmpty();
+    if (tree.modifiers().annotations().isEmpty()) {
```

Comment:
As per documentation of java stream api : https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#anyMatch-java.util.function.Predicate- : this is not required as anyMatch will return false is stream is empty.